.github/workflows/ci.yaml

```diff
@@ -0,0 +1,108 @@
+name: CI Pipeline
+
+on:
+  workflow_call:
+    secrets:
+      MAIL_USERNAME:
+        description: 'Username for the email notification step'
+        required: true
+      MAIL_PASSWORD:
+        description: 'Password for the email notification step'
+        required: true
+    inputs:
+      custom_docker_tag:
+        description: 'Custom Docker tag'
+        # If not specified, then the 'latest' tag will be used
+        default: ''
+        required: false
+        type: string
+
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+env:
+  DOCKER_REGISTRY: ghcr.io
+  DOCKER_REPO: angular-react-starter-image
+  DOCKER_TAG: ${{ inputs.custom_docker_tag || 'latest' }}
+  MAIL_SERVER_ADDRESS: smtp.gmail.com
+  MAIL_SERVER_PORT: 465
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
```

Comment:
make code consistent, so if you don't have empty rows in parent keys above, don't use it here. Or use it everywhere.